### PR TITLE
Add metadata annotations for UXP getting-started package

### PR DESCRIPTION
**Pull request by jastang**
Fixes https://github.com/upbound/squad-marketplace/issues/82

Signed-off-by: Jason Tang <jason@upbound.io>

### Description of your changes
Add metadata annotations for the UXP getting-started package for friendlier Marketplace consumption. Icon will be handled separately (https://github.com/upbound/squad-marketplace/issues/49).

Pushing to the public repo will have to be verified manually after this is approved (v0.0.6)

Fixes https://github.com/upbound/squad-marketplace/issues/82

I have:

- [x] Read and followed Upbound's [contribution process](https://git.io/fj2m9).
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR, as appropriate.

### How has this code been tested

Built and pushed xpkg to private repository to confirm annotations are rendered correctly in the UI.

https://marketplace.upbound.io/configurations/upbound/jason-getting-started-updates/v0.0.6

### CODE REVIEW

File: docs/getting-started/configuration/crossplane.yaml [hunk 1]
```diff
@@ -3,6 +3,16 @@ kind: Configuration
...
+    meta.crossplane.io/maintainer: Upbound <support@upbound.io>
+    meta.crossplane.io/source: github.com/upbound/universal-crossplane
+    meta.crossplane.io/license: Apache-2.0
+    meta.crossplane.io/description: |
+      An introductory example to UXP and Composition for AWS.
+    meta.crossplane.io/readme: |
+      An introductory example to UXP and Composition for AWS.
+      This will enable provisioning of an RDS database instance.
+
+      [Quickstart](https://docs.upbound.io/quickstart/community-providers/)
```
Comment (hasheddan) [inline docs/getting-started/configuration/crossplane.yaml:15]: @jastang looks like this quickstart involves installing a provider directly -- is that what we want to link to for this package?

Comment (jastang) [reply docs/getting-started/configuration/crossplane.yaml:15]: ah, good catch! This was just a placeholder. Looks like the original guide hasn't been moved to the new docs site yet. I'll follow up here

Comment (jastang) [reply docs/getting-started/configuration/crossplane.yaml]: The appropriate docs are only available in the archive for now according to @plumbis, so I'll remove this until they're available in docs.upbound.io

**Comment by jastang** [comment]
Closing in favour of #279.

**Pull request closed without merging**